src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/CustomConverterTests/CustomConverterTests.KeyConverters.cs

```diff
@@ -0,0 +1,167 @@
+        }
+
+        [Fact]
+        public static void ExtensionDataProperty_Serialization_IgnoreCustomStringKeyConverter()
+        {
+            var options = new JsonSerializerOptions { Converters = { new EmbeddedJsonKeyConverter<string>() } };
+            var value = new PocoWithExtensionDataProperty();
+
+            string expectedJson = @"{""key"":42}";
+            string json = JsonSerializer.Serialize(value, options);
+
+            Assert.Equal(expectedJson, json);
+        }
+
+        public class PocoWithExtensionDataProperty
+        {
+            [JsonExtensionData]
+            public Dictionary<string, object> Data { get; set; } = new Dictionary<string, object> { ["key"] = 42 };
+        }
+
+        [Theory]
+        [InlineData(InvalidCustomKeyConverter.InvalidOperationType.DoNothing)]
+        [InlineData(InvalidCustomKeyConverter.InvalidOperationType.HandleEntireParentObject)]
+        [InlineData(InvalidCustomKeyConverter.InvalidOperationType.HandleEntireProperty)]
+        public static void InvalidCustomKeyConverter_Serialization(InvalidCustomKeyConverter.InvalidOperationType invalidOperationType)
+        {
+            var options = new JsonSerializerOptions { Converters = { new InvalidCustomKeyConverter { Type = invalidOperationType } } };
+            var value = new Dictionary<string, int> { ["key"] = 42 };
+
+            Assert.Throws<JsonException>(() => JsonSerializer.Serialize(value, options));
+        }
+
+        [Theory]
+        [InlineData(InvalidCustomKeyConverter.InvalidOperationType.HandleEntireParentObject, typeof(JsonException))]
+        [InlineData(InvalidCustomKeyConverter.InvalidOperationType.HandleEntireProperty, typeof(JsonException))]
+        [InlineData(InvalidCustomKeyConverter.InvalidOperationType.ReturnNull, typeof(ArgumentNullException))]
+        public static void InvalidCustomKeyConverter_Deserialization(InvalidCustomKeyConverter.InvalidOperationType invalidOperationType, Type exceptionType)
+        {
+            var options = new JsonSerializerOptions { Converters = { new InvalidCustomKeyConverter { Type = invalidOperationType } } };
+            string json = @"{""key1"" : 1, ""key2"" : 2 }";
+
+            Assert.Throws(exceptionType, () => JsonSerializer.Deserialize<Dictionary<string, int>>(json, options));
+        }
+
+        public class EmbeddedJsonKeyConverter<T> : JsonConverter<T>
+        {
+            public override T Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions _)
+                => throw new NotSupportedException();
+
+            public override void Write(Utf8JsonWriter writer, T value, JsonSerializerOptions _)
+                => throw new NotSupportedException();
+
+            public override void WriteAsPropertyName(Utf8JsonWriter writer, T value, JsonSerializerOptions _)
+                => writer.WritePropertyName(JsonSerializer.Serialize(value));
```

Comment:
This is the same behavior for primitives as the default STJ behavior, so it would be hard to verify that the converter is being called. However, I see the test data has arrays etc so we should be good.

Reply:
Had similar thoughts, but even for string types the payload incorporates escaped quotes. Should be ok.